common/test/views/support/cleaner/AmpEmbedCleanerTest.scala

```diff
@@ -193,9 +193,22 @@ class AmpEmbedCleanerTest extends FlatSpec with Matchers {
     result.getElementsByTag("amp-facebook").attr("layout") should be("responsive")
   }
 
+  "AmpEmbedCleaner" should "replace a facebook non-guardian video embed with a valid amp-facebook video embed" in {
+    val facebookOrganisationId = "Channel4"
+    val faceookVideoId = "10154084521542330"
+    val facebookVideoUrl = s"https://www.facebook.com/$facebookOrganisationId/videos/$faceookVideoId/"
+    val result = cleanDocumentWithVideos(facebookVideoUrl)
+    result.getElementsByTag("amp-facebook").size should be(1)
+    result.getElementsByTag("amp-facebook").attr("data-href") should be(s"https://www.facebook.com/$facebookOrganisationId/videos/$faceookVideoId")
+    result.getElementsByTag("amp-facebook").attr("data-embed-as") should be("video")
+    result.getElementsByTag("amp-facebook").attr("width") should be("5")
+    result.getElementsByTag("amp-facebook").attr("height") should be("3")
+    result.getElementsByTag("amp-facebook").attr("layout") should be("responsive")
+  }
+
+
   "AmpEmbedCleaner" should "not replace an iframe in a fake Facebook video-element with an amp-facebook element" in {
     val result = cleanDocumentWithVideos(
-      "https://www.facebook.com/nottheguardian/videos/123456",
       "https://facbook.com.zz/theguardian/123456/",
```

Comment:
I don't think you really have to so extensively test your regex! But I suppose it does not do any harm